### PR TITLE
Remove most usages of 1-arg Script ctor

### PR DESCRIPTION
**Pull request by nik9000**
The one argument ctor for `Script` creates a script with the
default language but most usages of are for testing and either
don't care about the language or are for use with
`MockScriptEngine`. This replaces most usages of the one argument
ctor on `Script` with calls to `ESTestCase#mockScript` to make
it clear that the tests don't need the default scripting language.

I've also factored out some copy and pasted script generation
code into a single place. I would have had to change that code
to use `mockScript` anyway, so it was easier to perform the
refactor.

Relates to #16314

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/search/aggregations/BaseAggregationTestCase.java [hunk 1]
```diff
@@ -176,4 +177,22 @@ public String randomNumericField() {
...
+            break;
+        case 1:
+            factory.field(field);
+            factory.script(mockScript("_value + 1"));
+            break;
+        case 2:
+            factory.script(mockScript("doc[" + field + "] + 1"));
+            break;
+        default:
+            throw new UnsupportedOperationException("Unknow random operation [" + choice + "]");
```
Comment (rjernst) [inline core/src/test/java/org/elasticsearch/search/aggregations/BaseAggregationTestCase.java]: I would make this an AssertionError, it should be impossible.

**Review by rjernst — APPROVED** [review]
Lgtm

File: core/src/test/java/org/elasticsearch/search/aggregations/BaseAggregationTestCase.java (hunk 1 above)
Comment (nik9000) [inline core/src/test/java/org/elasticsearch/search/aggregations/BaseAggregationTestCase.java]: Sure. Will fix and merge.

**Comment by nik9000** [comment]
Thanks for reviewing @rjernst! I've merged to master and I'll backport to 5.x.

**Pull request merged**